applications/StructuralMechanicsApplication/tests/cpp_tests/test_tangent_tensor_numerical_derivation_utility.cpp

```diff
@@ -16,106 +17,145 @@
+                KRATOS_WARNING("LinearElasticCasePertubationTensorUtility") << "Be careful tangent_moduli(" << i << " ," << j << ") is greater tha 0: " <<  tangent_moduli(i, j) << std::endl;
+            }
+        }
+    }
 }
 
 /**
-    * Check the correct calculation of the uniaxial stress of the yield surfaces
-    */
-KRATOS_TEST_CASE_IN_SUITE(PertubationTensorTestUtility, KratosStructuralMechanicsFastSuite)
+ * @brief This test tests that the perturbation utility is valid for computing the hyperelastic tensor
+ */
+KRATOS_TEST_CASE_IN_SUITE(HyperElasticCasePertubationTensorUtility, KratosStructuralMechanicsFastSuite)
 {
-    ConstitutiveLaw::Parameters rValues;
-    Properties rMaterialProperties;
-    Vector rStressVector, rStrainVector;
+    ConstitutiveLaw::Parameters cl_configuration_values;
+    Properties material_properties;
+    Vector stress_vector, strain_vector;
     ProcessInfo CurrentProcessInfo;
 
-    Flags ConstitutiveLawOptions;
-    ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
-    ConstitutiveLawOptions.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
-    ConstitutiveLawOptions.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
-
-    rMaterialProperties.SetValue(YOUNG_MODULUS, 210e9);
-    rMaterialProperties.SetValue(POISSON_RATIO, 0.22);
-
-    rStressVector = ZeroVector(6);
-    rStressVector[0] = 5.40984e+06;
-    rStressVector[1] = 5.40984e+06;
-    rStressVector[2] = 1.91803e+07;
-    rStressVector[3] = 0.0;
-    rStressVector[4] = 0.0;
-    rStressVector[5] = 1.45804e-10;
-
-    rStrainVector = ZeroVector(6);
-    rStrainVector[0] = 0.0;
-    rStrainVector[1] = 0.0;
-    rStrainVector[2] = 8.0e-5;
-    rStrainVector[3] = 0.0;
-    rStrainVector[4] = 0.0;
-    rStrainVector[5] = 1.6941e-21;
-
-    Matrix F = IdentityMatrix(6);
-
-    rValues.SetMaterialProperties(rMaterialProperties);
-    rValues.SetDeformationGradientF(F);
-    rValues.SetStrainVector(rStrainVector);
-    rValues.SetStressVector(rStressVector);
-    rValues.SetOptions(ConstitutiveLawOptions);
-    rValues.SetProcessInfo(CurrentProcessInfo);
-
-    ElasticIsotropic3D ConstitutiveLaw = ElasticIsotropic3D();
-    ElasticIsotropic3D *pConstitutiveLaw = &ConstitutiveLaw;
+    Flags constitutive_law_options;
+    constitutive_law_options.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
+    constitutive_law_options.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, false);
+    constitutive_law_options.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
+
+    material_properties.SetValue(YOUNG_MODULUS, 210e9);
+    material_properties.SetValue(POISSON_RATIO, 0.22);
+
+    stress_vector = ZeroVector(6);
+    stress_vector[0] = 5.40984e+06;
+    stress_vector[1] = 5.40984e+06;
+    stress_vector[2] = 1.91803e+07;
+    stress_vector[5] = 1.45804e-10;
+
+    strain_vector = ZeroVector(6);
+    strain_vector[2] = 8.0e-5;
+    strain_vector[5] = 1.6941e-21;
+
+    Matrix deformation_gradient = ZeroMatrix(3, 3);
```

Comment:
I think that F was a better name for this (as it was called before)

Reply:
F is the deformation gradient, I can call it deformation_gradient_F, but I like descriptive names, no names that require to know the notation

Reply:
well in the doc of the CL (and in essentially all books of computational mechanics) it is called F

Reply:
I will use deformation_gradient_F. i don't like variables with very shost names that later no body knows what is. Just now we have a discussion with @marandra for a mu parameter that was the Lame mu parameter, but we confused with Poisson